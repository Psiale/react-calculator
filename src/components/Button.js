import PropTypes from 'prop-types';
import React from 'react';

const Button = ({ name, clickHandler }) => {
  const handleClick = name => (clickHandler(name));
  return (
    <>
      <button onClick={handleClick(name)} type="button">{name}</button>
    </>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
