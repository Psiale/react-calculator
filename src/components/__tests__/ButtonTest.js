import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../Button';

configure({ adapter: new Adapter() });

describe('Button Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Button name="0" />).toJSON;
    expect(tree).toMatchSnapshot();
  });

  it('should have the name property displayed inner content ', () => {
    const wrapper = shallow(<Button name="0" />);
    const text = wrapper.text();
    expect(text).toContain('0');
  });
});
