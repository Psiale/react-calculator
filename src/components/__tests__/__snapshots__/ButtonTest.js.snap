// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches the snapshot 1`] = `[Function]`;
